project/frontend/src/containers/Dashboard/Dashboard.js

```diff
@@ -56,56 +57,43 @@ const Dashboard = (props) => {
     },
   ];
 
-  const placeholderTrends = [
-    "Kellyanne Conway",
-    "Filibuster",
-    "Fultondale al",
-    "Ben Askren",
-    "Stock-market",
-    "Etsy stock",
-    "Elliot Page",
-    "Susan Rice",
-    "Raya and the Last Dragon",
-    "Starlink",
-  ];
+  const toDayHourMinute = (totalTime) => {
+    let millisInDay = 24 * 60 * 60 * 1000,
+      millisInHour = 60 * 60 * 1000,
+      days = Math.floor(totalTime / millisInDay),
+      hours = Math.floor((totalTime - days * millisInDay) / millisInHour),
+      minutes = Math.round(
+        (totalTime - days * millisInDay - hours * millisInHour) / 60000
+      ),
+      pad = function (n) {
+        return n < 10 ? "0" + n : n;
+      };
+    if (minutes === 60) {
+      hours++;
+      minutes = 0;
+    }
+    if (hours === 24) {
+      days++;
+      hours = 0;
+    }
+    return [days, pad(hours), pad(minutes)];
+  };
 
-  const placeholderRecentBuys = [
-    {
-      buyerName: "Mercury Lin",
-      buyerEmail: "mercurylin@google.com",
-      amountBought: 3921,
-      searchTerm: "pineapples",
-      timeBought: 1580085733000,
-    },
-    {
-      buyerName: "Emma Hogan",
-      buyerEmail: "emmahogan@google.com",
-      amountBought: 64526,
-      searchTerm: "bitcoin",
-      timeBought: 1580129754000,
-    },
-    {
-      buyerName: "Phoebe Khokgawe",
-      buyerEmail: "phoebek@google.com",
-      amountBought: 4213,
-      searchTerm: "chicken wings",
-      timeBought: 1579664662000,
-    },
-    {
-      buyerName: "Tex Moran",
-      buyerEmail: "texm@google.com",
-      amountBought: 1,
-      searchTerm: "kangaroo",
-      timeBought: 1611877597799,
-    },
-    {
-      buyerName: "Leon Nemets",
-      buyerEmail: "nemleon@google.com",
-      amountBought: "777",
-      searchTerm: "rick roll",
-      timeBought: 1580214808000,
-    },
-  ];
+  const formatDHM = (dhm) => {
+    return dhm[0] + "  Days  " + dhm[1] + "  Hours  " + dhm[2] + "  Minutes  ";
+  };
+
+  const getTimeRemaining = () => {
+    const millisNow = Date.now();
+    const remainingTime = props.generalInfo.endDate - millisNow;
+    return formatDHM(toDayHourMinute(remainingTime));
+  };
+
+  const getOrdinalOnly = (n) => {
+    const s = ["th", "st", "nd", "rd"],
+      v = n % 100;
+    return s[(v - 20) % 10] || s[v] || s[0];
+  };
```

Comment:
I have seen this one before, can it be factored out?

Reply:
Hmm, I did think about that, I'm just not quite sure where to put it

Reply:
I guess I could make a Rank component?

Reply:
Made a Rank component. In hindsight even though it's like a 2 line component, that was a really good idea.